src/lib/middleware/rateLimiting.ts

```diff
@@ -0,0 +1,48 @@
+import config from "../../config"
+import { RateLimiterMemory, RateLimiterRedis } from "rate-limiter-flexible"
+
+const {
+  OPENREDIS_URL,
+  REQUEST_EXPIRES,
+  REQUEST_LIMIT,
+  REQUEST_PER_INSTANCE_FALLBACK,
+} = config
+
+export const rateLimiterMiddlewareFactory = redisClient => {
+  /**
+   * Used as a per process limiter if for whatever reason
+   * redis becomes unavailable
+   */
+  const rateLimiterMemory = new RateLimiterMemory({
+    points: REQUEST_PER_INSTANCE_FALLBACK,
+    duration: REQUEST_EXPIRES,
+  })
+
+  let rateLimiter = rateLimiterMemory
+
+  if (OPENREDIS_URL && redisClient) {
+    rateLimiter = new RateLimiterRedis({
+      storeClient: redisClient,
+      points: REQUEST_LIMIT,
```

Comment:
"points" 🤔? 

Is that meant to be more abstract than # of requests, like you could weight requests differently?

Reply:
Yep, it's not a 1-to-1 mapping to requests. The consume method takes an argument of how many points to consume. 

An example from their readme:

```js
rateLimiter.consume(remoteAddress, 2) // consume 2 points
    .then((rateLimiterRes) => {
      // 2 points consumed
    })
    .catch((rateLimiterRes) => {
      // Not enough points to consume
    });
```

In this case they're consuming 2 "points" per the request. For our case we're consuming a point on every request, but that isn't always the case.